src/System.Private.ServiceModel/tools/test/Bridge/Bridge/IdleTimeoutManager.cs

```diff
@@ -55,6 +58,26 @@ public IdleTimeoutManager(TimeSpan idleTimeout)
                 _waitTimeout = idleTimeout;
                 _timer = new Timer(this.TimeOutCallback);
                 this.Restart();
+
+                // Register to be notified when the Bridge configuration changes
+                ConfigController.BridgeConfigurationChanged += (object s, Tuple<BridgeConfiguration, BridgeConfiguration> tuple) =>
+                {
+                    BridgeConfiguration newConfig = tuple.Item2;
+
+                    int minutes = newConfig.BridgeIdleTimeoutMinutes;
+                    TimeSpan newTimeout = (minutes <= 0)
```

Comment:
I changed the timeout to be a TimeSpan to eliminate any special values that receive different treatment.  You can now shut down the Bridge in 5 seconds by sending it a timeout of "00:00:05" or ask it to run for days.